bin/varnishd/cache/cache_session.c

```diff
@@ -104,9 +105,9 @@ ses_set_attr(const struct sess *sp, enum sess_attr a, const void *src, int sz)
 	AN(src);
 	assert(sz > 0);
 
-	if (sp->sattr[a] == 0xffff)
+	if (sp->sattr[a] == SES_NOATTR_OFFSET)
 		return (-1);
-	dst = sp->ws->s + sp->sattr[a];
+	dst = WS_AtOffset(sp->ws, sp->sattr[a], sz);
```

Comment:
As this is the only use case of the rather obscure offset magic, I wonder if we should rather afford some more bytes and use pointers for the session attrs? @bsdphk ?

Reply:
second thought: it's quite some bytes for someone with en efficiency fetish, so I we should keep it...